pkg/controller/status/status.go

```diff
@@ -129,8 +131,22 @@ func (c *Controller) merge(existing *configv1.ClusterOperator) *configv1.Cluster
 			klog.Errorf("Programmer error: status source %d %T reported an empty message: %#v", i, source, summary)
 			continue
 		}
-		reason = summary.Reason
-		errors = append(errors, summary.Message)
+
+		switch source.(type) {
+		case *controllerstatus.UploaderStatus:
+			uploadErrorReason = summary.Reason
+			uploadErrorMessage = summary.Message
+			// NotAuthorized is a special case where we want to disable the operator
+			if summary.Reason == "NotAuthorized" {
```

Comment:
having multiple sets of "uploadError* / disabled*" is going to be confusing, if you add an operation field to the struct use `uploadReason` `uploadMessage` instead.

Reply:
It doesn't fit very well, as there are other reasons for disabling (https://github.com/openshift/insights-operator/pull/16/files#diff-bf2e5e59fb3ef5c42d555b1b4da9c210R168), that are not related to `uploadError*`.

And I don't want to reuse `reason` either, as you might have other reason for `degraded` state, and other for `disabled` or `uploading`

Reply:
make this a helper `IsNotAuthorizedReason(summary.Reason)`

Reply:
updated